x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/execution/search/extractor/FieldHitExtractorTests.java

```diff
@@ -336,6 +336,24 @@ public void testFieldWithDotsAndSamePathButDifferentHierarchy() {
         assertThat(ex.getMessage(), is("Multiple values (returned by [a.b.c.d.e.f.g]) are not supported"));
     }
 
+    public void testObjectsForSourceValue() throws IOException {
+        String fieldName = randomAlphaOfLength(5);
+        FieldHitExtractor fe = new FieldHitExtractor(fieldName, null, false);
+        SearchHit hit = new SearchHit(1);
```

Comment:
Why did you choose to call `extract(hit)` and not follow the approach as here for example: https://github.com/elastic/elasticsearch/pull/37502/files#diff-9aaee9be08445653bba7407b9f2b5ca3R302

Reply:
No particular reason. Both approaches would work I think. I can replace it if you prefer.

Reply:
Nevermind, just wanted to check if there was a particular reason.